javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -335,33 +335,119 @@ public static boolean containsSameByName(
+        if (av1 == av2) {
+            return 0;
+        } else if (av1 == null) {
+            return -1;
+        } else if (av2 == null) {
+            return 1;
+        }
+        return compareAnnotationValueValue(av1.getValue(), av2.getValue());
+    }
+
+    /**
+     * Return 0 if the two annotation values are the same.
+     *
+     * @param val1 a value returned by {@code AnnotationValue.getValue()}
+     * @param val2 a value returned by {@code AnnotationValue.getValue()}
+     */
+    private static int compareAnnotationValueValue(@Nullable Object val1, @Nullable Object val2) {
+        if (val1 == val2) {
+            return 0;
+        } else if (val1 == null) {
+            return -1;
+        } else if (val2 == null) {
+            return 1;
+        }
+        // Can't use deepEquals() to compare val1 and val2, because they might have mismatched
+        // AnnotationValue vs. CheckerFrameworkAnnotationValue, and AnnotationValue doesn't override
+        // equals().  So, write my own version of deepEquals().
+        if ((val1 instanceof List<?>) && (val2 instanceof List<?>)) {
+            List<?> list1 = (List<?>) val1;
+            List<?> list2 = (List<?>) val2;
+            if (list1.size() != list2.size()) {
+                return list1.size() - list2.size();
+            }
+            // Don't compare setwise, because order can matter. These mean different things:
+            //   @LTLengthOf(value={"a1","a2"}, offest={"0", "1"})
+            //   @LTLengthOf(value={"a2","a1"}, offest={"0", "1"})
+            for (int i = 0; i < list1.size(); i++) {
+                Object v1 = list1.get(i);
+                Object v2 = list2.get(i);
+                int result = compareAnnotationValueValue(v1, v2);
+                if (result != 0) {
+                    return result;
+                }
+            }
+            return 0;
+        } else if ((val1 instanceof AnnotationMirror) && (val2 instanceof AnnotationMirror)) {
+            return compareAnnotationMirrors((AnnotationMirror) val1, (AnnotationMirror) val2);
+        } else if ((val1 instanceof AnnotationValue) && (val2 instanceof AnnotationValue)) {
+            // This case occurs because of the recursive call when comparing arrays of
+            // annotation values.
+            return compareAnnotationValue((AnnotationValue) val1, (AnnotationValue) val2);
+        }
+
+        if ((val1 instanceof Type.ClassType) && (val2 instanceof Type.ClassType)) {
+            // Type.ClassType does not override equals
+            if (TypesUtils.areSameDeclaredTypes((Type.ClassType) val1, (Type.ClassType) val2)) {
+                return 0;
+            }
+        }
+        if (Objects.equals(val1, val2)) {
+            return 0;
+        }
+        int result = val1.toString().compareTo(val2.toString());
+        if (result == 0) {
+            result = -1;
+        }
+        return result;
+    }
```

Comment:
Nit: Empty line between top-level elements.

Reply:
This is fixed.